src/Microsoft.Identity.Web/TokenCacheProviders/Distributed/MsalDistributedTokenCacheAdapterOptions.cs

```diff
@@ -12,5 +12,17 @@ namespace Microsoft.Identity.Web.TokenCacheProviders.Distributed
     /// </summary>
     public class MsalDistributedTokenCacheAdapterOptions : DistributedCacheEntryOptions
     {
+        /// <summary>
+        /// In memory (L1) cache size limit in Mb.
+        /// Default is 500 Mb.
+        /// </summary>
+        public long L1CacheSizeLimit { get; set; } = 500;
+
+        /// <summary>
+        /// Value more than 0, less than 1, to set the In Memory (L1) cache
+        /// expiration time values relative to the Distributed (L2) cache.
+        /// Default is 1.
+        /// </summary>
+        public double L1ExpirationTimeRatio { get; set; } = 1;
```

Comment:
I know this is used in MISE, I'm not convinced about this time ratio. would be good with an example in the summary which explains what it means for the L2 cache size in actual numbers.

Reply:
It's actually not using MISE, we invented this. will add more details.

---
In reply to: [586554302](https://github.com/AzureAD/microsoft-identity-web/pull/1037#discussion_r586554302) [](ancestors = 586554302)

Reply:
I don't think we should expose this

Reply:
Why @henrik-me : don't we want the L1 cache to expire before the L2 one ?

Reply:
I made it internal for now. [https://github.com/AzureAD/microsoft-identity-web/pull/1039](https://github.com/AzureAD/microsoft-identity-web/pull/1039)

---
In reply to: [586726432](https://github.com/AzureAD/microsoft-identity-web/pull/1037#discussion_r586726432) [](ancestors = 586726432)

Reply:
BTW, @henrik-me : this was about time, not size. There are 2 dimensions in these caches: the freshness, and the size limit.